packages/compile-solidity/src/compilerSupplier/index.ts

```diff
@@ -0,0 +1,117 @@
+import path from "path";
+import fs from "fs";
+import semver from "semver";
+
+import { Docker, Local, Native, VersionRange } from "./loadingStrategies";
+
+const defaultSolcVersion = "0.5.16";
+
+export class CompilerSupplier {
+  private events: any;
+  private version: string;
+  private docker: boolean;
+  private compilerRoots: string[];
+  private strategyOptions: Partial<{
+    version: string;
+    docker: boolean;
+    compilerRoots: string[];
```

Comment:
We opened [this documentation issue](https://github.com/trufflesuite/trufflesuite.com/issues/1043) to ensure this knowledge is available to our users.

Reply:
I posit that, since there's an open issue, this needn't block merge of this PR.

Reply:
Makes sense, that issue's on a separate repo anyway.